### ISSUE TITLE
problem connecting new agent

### ISSUE THREAD

**Issue by d2-d2** [s3inlc/hashtopussy#121]
HI,

I'm using latest github sources of hashtopussy, setup all according to docs and my knowledge when playing with hashtopus. Now the problem I have right now is with hashtopussy.exe which can't connect to server. I'm providing URL https://127.0.0.1/hashtopussy/src/api/server.php and even though connection is being closed. Is it because SSL and self-signed certificate? Is there a version of hashtopussy.exe that doesn't require/force SSL? I'm testing this setup on XAMPP.

**Comment by blazer2x** [comment]
Have you tried and put non https connect? like http://127.0.0.1/hashtopussy/src/api/server.php

**Comment by d2-d2** [comment]
Yes, and actually this is another thing I wanted to let you know about. I was using http and https connection to control panel and it seems that there is a confusion on adding agent page:

```
http://localhost/hashtopussy/src/agents.php?new=true
```

Here is what it says:
```If you are asked to provide the API url on the client to connect to, you need to enter the following:
https://localhost:80/hashtopussy/src/api/server.php
```

Now I've downloaded hashtopussy.exe and tried registering agent and here is what it says in console:

```
Client Version 0.38
Please enter server connect URL (https will be used unless specified):
(1) localhost:80/hashtopussy/src/api/server.php
Testing connection to (2) https://localhost:80/hashtopussy/src/api/server.php
{"action":"test"}
```

Please not that I'm providing http (1) and then agent is messing with my input (2) around trying to access SSL over port 80/tcp.

I'd say something is not right with registration process.

**Comment by blazer2x** [comment]
try change the url file or enter this
**http**://localhost/hashtopussy/src/api/server.php

If you do not want to use https  you must specify http in the url

**Comment by d2-d2** [comment]
Unfortunately, it still chose to use HTTPS. I've done what you asked:
```
http://localhost/hashtopussy/src/api/server.php
```

And the response is that URL was validated, but agent can't be downloaded:
```
c:\Users\d2-d2\Downloads>hashtopussy.exe
Client Version 0.38
Please enter server connect URL (https will be used unless specified):
http://localhost/hashtopussy/src/api/server.php
Testing connection to http://localhost/hashtopussy/src/api/server.php
{"action":"test"}
{"action":"test","response":"SUCCESS"}
Checking for client updates
{"action":"update","type":"csharp","version":"0.38"}
{"action":"update","response":"SUCCESS","version":"NEW","url":"https:\/\/localhost:80\/hashtopussy\/src\/agents.php?download=1"}
Downloading from https://localhost:80/hashtopussy/src/agents.php?download=1

Wyjątek nieobsłużony: System.Net.WebException: Connection closed. ---> System.IO.IOException:
   w System.Net.Security.SslState.StartReadFrame(Byte[] buffer, Int32 readBytes, AsyncProtocolRequest asyncRequest)
   w System.Net.Security.SslState.StartReceiveBlob(Byte[] buffer, AsyncProtocolRequest asyncRequest)
   w System.Net.Security.SslState.ForceAuthentication(Boolean receiveFirst, Byte[] buffer, AsyncProtocolRequest asyncRequest)
   w System.Net.Security.SslState.ProcessAuthentication(LazyAsyncResult lazyResult)
   w System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   w System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   w System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   w System.Net.TlsStream.ProcessAuthentication(LazyAsyncResult result)
   w System.Net.TlsStream.Write(Byte[] buffer, Int32 offset, Int32 size)
   w System.Net.ConnectStream.WriteHeaders(Boolean async)
   --- ### ---
   w System.Net.WebClient.DownloadFile(Uri address, String fileName)
   w hashtopussy.downloadClass.DownloadFile(String urlAddress, String location)
   w hashtopussy.updateClass.runUpdate()
   w hashtopussy.Program.Main(String[] args)
```

**Comment by s3inlc** [comment]
There is a bug in the current code where it will always say it's https, this also affects the downloading of agents as the server build the url the same way like shown on the page.

Please try to use the dev/v0.2.0 branch, there this bug is fixed and you should have not problems in using it with http.

**Comment by d2-d2** [comment]
Actually, I diffed inc/Util.class.php and changed https to http. At least this is what I see was done in DEV branch.

Another problem arised. Client connects to server OK, and see that there is an update available. Downloading it, saving and ... process repeat itself. It's a infinite loop caused by incorrect setting in COnfig->Agent Binaries where agent version is 0.34 where hashtopussy.exe binary is marked as being at version 0.38. I've corrected that in my panel and it's now passing this step.

Next issue: I have to manually install hashcat as there seems to be no package included.

**Comment by s3inlc** [comment]
Thanks for the notification, it seems that I forgot to adjust to the newest version when I updated the binary.

We currently are discussing if we should add a default binary entry, but as we now support also OS X this is problematic because hashcat doesn't provide prebuilt OS X binaries.

**Comment by d2-d2** [comment]
I think anybody can download hashcat version for themselves. The only thing is to tell ppl where to upload it in panel and just notify users that binary is not included.

I have another small issue with tasks. When creating new one, htp is displaying this:

```
Use #HL# for hash list and assume all files in current directory.
```

but when it try to measure keyspace it can't find required files (wordlists, rules etc).

**Comment by winxp5421** [comment]
Binaries are not and probably will never be included with hashtopussy. People want the newest binaries the hashtopussy server only points directly to hashcat.net's pre-compiled binaries. To change where your install pulls these from navigate to "Hashcat > New Hashcat Release"

As for your Issue with wordlists / rules. make sure you have these files selected on the right hand side of the screen when creating a task.

Most of this information is located in the wiki.

**Comment by d2-d2** [comment]
I've read wiki and followed it to get this. When I assign wordlists they are assigned as-is. No subdirectories, nothing.

So to solve it obviously I had to use full path to rules and dictionaries. Now once that is finished another problem arised:

``
Server has requested the client to measure the keyspace for this task
Using c:\Users\d2-d2\Downloads\hp\files\ as working directory
c:\Users\d2-d2\Downloads\hp\hashcat\hashcat64.exe --hash-type=2811 --gpu-temp-retain 50 -w4 --potfile-disable --rules-file=c:\Users\d2-d2\Downloads\hp\hashcat\rules\best64.rule  c:\Password\_dic\_moved_best\0duplicates --session=hashtopussy --keyspace --quiet
{"action":"keyspace","token":"lheibjOyP4","taskId":7,"keyspace":44931640}
{"action":"chunk","token":"lheibjOyP4","taskId":7}
{"action":"chunk","response":"SUCCESS","status":"benchmark"}

Using c:\Users\d2-d2\Downloads\hp\files\ as working directory
c:\Users\d2-d2\Downloads\hp\hashcat\hashcat64.exe --hash-type=2811 --gpu-temp-retain 50 -w4 --potfile-disable --rules-file=c:\Users\d2-d2\Downloads\hp\hashcat\rules\best64.rule "c:\Users\d2-d2\Downloads\hp\hashlists\1" c:\Password\_dic\_moved_best\0duplicates --runtime=30 --restore-disable --potfile-disable --machine-readable --session=hashtopussy --weak=0 --progress-only
Server has requested the client to perform a speed benchmark
{"action":"bench","token":"lheibjOyP4","taskId":7,"type":"speed","result":"25763544:522386,39"}
{"action":"bench","response":"ERROR","message":"Invalid benchmark result!"}
ERROR
Invalid benchmark result!
Server rejected benchmark
Check the hashlist was downloaded correctly

c:\Users\d2-d2\Downloads\hp>
```

At this point, I have no idea what that mean and why is it happening. Any hints?

BTW - I'm not sure if this still count for original issue, maybe I should open another one? Or perhaps we can talk on IRC? I'm sitting on #hashkiller.

**Comment by winxp5421** [comment]
Just out of curiosity where did you enter "--gpu-temp-retain 50 -w4" ?

**Comment by d2-d2** [comment]
Tasks->New Task->Command LIne:

**Comment by winxp5421** [comment]
"--gpu-temp-retain 50 -w4" are agent specific parameters and should be added in each client respectively under Agents > Show Agents > "select agent name"

can you post the exact task from hashtopussy.

have you uploaded your wordlists to hashtopussy?

**Comment by s3inlc** [comment]
I'm closing this issue now, the problems were all solved by the discussion on IRC

**Comment by Ode55a2000** [comment]
Hi. I'm install hashtopolis and have same problem.

> 

Invalid benchmark result!
Server rejected benchmark
Check the hashlist was downloaded  correctly

> 
Please tel me design.
Sorry for my english

**Comment by s3inlc** [comment]
Hey, this is a closed issue, your problem sounds more general and lacks some descriptions of what you've tried to run etc.

I recommend you use the forum at https://hashtopolis.org to ask for help and providing some information what exactly failed.
